app/recurring_job.go

```diff
@@ -445,7 +445,12 @@ func (job *Job) doSnapshotCleanup(backupDone bool) (err error) {
 		return err
 	}
 
-	cleanupSnapshotNames := job.listSnapshotNamesToCleanup(collection.Data, backupDone)
+	allowBackupSnapshotDeleted, err := job.GetSettingAsBool(string(types.SettingNameAutoCleanupRecurringJobBackupSnapshot))
```

Comment:
Any reason for parsing through as argument instead of getting this setting directly within the  `filterExpiredSnapshotsOfCurrentRecurringJob` method?

Reply:
IIRC it can not return an error in then `filterExpiredSnapshotsOfCurrentRecurringJob` method.
Moved into the `filterExpiredSnapshotsOfCurrentRecurringJob` method and changed to leave a message if an error occurred from the `job.GetSettingAsBool` method.
Thanks.